controllers/profile.js

```diff
@@ -0,0 +1,34 @@
+const Profile = require("../models/Profile");
+var ObjectId = require("mongoose").Types.ObjectId;
+const { StatusCodes } = require("http-status-codes");
+const { BadRequestError, UnauthenticatedError } = require("../errors");
+
+const getProfileData = async (req, res) => {
+  try {
+    const data = await Profile.findOne({ email: req.body.email });

```

Comment:
While not impossible, GET requests don't usually have a body and you probably don't need one. You could get the email of the user that made the request (using the token) or you could get this parameter from other part of the request (a header, the query or the route).
Take a look at [this](https://www.baeldung.com/cs/http-get-with-body).

Reply:
Thank you, Gabriel. I felt uncomfortable while writing the request 😊